core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingInterceptor.java

```diff
@@ -129,20 +132,20 @@ private boolean performPartitionCheck(InvocationContext ctx, LocalFlagAffectedCo
    }
 
    private CompletableFuture<Void> handleDataReadCommand(InvocationContext ctx, DataCommand command) {
-      Object key = command.getKey();
       return ctx.onReturn((rCtx, rCommand, rv, throwable) -> {
+         DataCommand dataCommand = (DataCommand) rCommand;
          if (throwable != null) {
-            if (throwable instanceof RpcException && performPartitionCheck(rCtx, ((DataCommand) rCommand))) {
+            if (throwable instanceof RpcException && performPartitionCheck(rCtx, dataCommand)) {
                // We must have received an AvailabilityException from one of the owners.
                // There is no way to verify the cause here, but there isn't any other way to get an invalid
                // get response.
-               throw log.degradedModeKeyUnavailable(key);
+               throw log.degradedModeKeyUnavailable(dataCommand.getKey());
```

Comment:
The changes in this PR are removing a special check when the result is `null`, because this is a valid value; "If all owners left" (as per the comment), the command should throw an exception, but never return `null` (when a null value was not read on proper owner). So I believe that the condition below cannot be true (without catching the `RpcException` earier on).
When the check has been add, it was a non-systematic workaround that fixed the problem on a wrong place.